frontend/src/metabase/query_builder/components/view/ViewFooter/ViewFooterViewControl.tsx

```diff
@@ -0,0 +1,113 @@
+import { useMemo } from "react";
+import { t } from "ttag";
+
+import { EditorViewControl } from "embedding-sdk/components/private/EditorViewControl";
+import { useDispatch, useSelector } from "metabase/lib/redux";
+import { isNotNull } from "metabase/lib/types";
+import {
+  setQueryBuilderMode,
+  setUIControls,
+} from "metabase/query_builder/actions";
+import {
+  getIsActionListVisible,
+  getIsShowingRawTable,
+  getIsVisualized,
+} from "metabase/query_builder/selectors";
+import { Icon, Tooltip } from "metabase/ui";
+import { getIconForVisualizationType } from "metabase/visualizations";
+import * as Lib from "metabase-lib";
+import type Question from "metabase-lib/v1/Question";
+
+interface ViewFooterControlProps {
+  question: Question;
+  isNotebook?: boolean;
+}
+
+const ViewFooterControl = ({
+  question,
+  isNotebook = false,
+}: ViewFooterControlProps) => {
+  const { isNative, isEditable } = Lib.queryDisplayInfo(question.query());
+  const isActionListVisible = useSelector(getIsActionListVisible);
+  const shouldShowEditorButton =
+    !isNative && isEditable && !question.isArchived() && isActionListVisible;
+  const vizIcon = getIconForVisualizationType(question.display());
+  const isShowingRawTable = useSelector(getIsShowingRawTable);
+  const isVisualized = useSelector(getIsVisualized);
+
+  const dispatch = useDispatch();
+
+  const data = useMemo(
+    () =>
+      [
+        shouldShowEditorButton
+          ? {
+              value: "editor",
+              label: (
+                <Tooltip label={t`Editor`}>
+                  <Icon
+                    aria-label={t`Switch to editor`}
+                    name="notebook"
+                    onClick={() => {
+                      dispatch(setQueryBuilderMode("notebook"));
+                    }}
```

Comment:
:x: We should be using `onChange ` prop from `SegmentedControl`.

This `onClick`:
- makes only the icon clickable, even though `cursor: pointer` is displayed on its parent ([video](https://github.com/user-attachments/assets/93681c3b-3f98-4bde-aef3-5d5597fbe403))
- breaks a11y (trying to use this control with keyboard, selecting does not work - [video](https://github.com/user-attachments/assets/304bc6df-47b9-44d1-adb8-d7d321334581))

Reply:
good point

Reply:
fixed